### PR TITLE
PMM-7 pmm client setup refactored

### PR DESCRIPTION
**Pull request by vyarosh**

### CODE REVIEW

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-3139-4113992
client docker - perconalab/pmm-client-fb:PR-3139-4113992
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-3139-4113992.tar.gz
Create Staging Instance: https://pmm.cd.percona.com/job/aws-staging-start/parambuild/?DOCKER_VERSION=perconalab/pmm-server-fb:PR-3139-4113992&CLIENT_VERSION=https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-3139-4113992.tar.gz

**Comment by vyarosh** [comment]
https://github.com/Percona-Lab/pmm-submodules/actions/runs/4443747715 - dev latest
https://github.com/Percona-Lab/pmm-submodules/actions/runs/4443750084 - FB tarball

File: .github/workflows/pmm2-testsuite.yml [hunk 1]
```diff
@@ -127,7 +131,39 @@ jobs:
...
+            curl -O https://raw.githubusercontent.com/Percona-QA/package-testing/PMM-7-tarball-improve/scripts/pmm2_client_install_tarball.sh
+            sudo bash -x pmm2_client_install_tarball.sh -v ${{ env.CLIENT_VERSION }}
+            echo 'Setup local pmm-agent...'
+            sudo pmm-agent setup --force --config-file=/usr/local/percona/pmm2/config/pmm-agent.yaml --server-address=127.0.0.1:443 --server-insecure-tls --server-username=admin --server-password=admin
+            sudo pmm-agent --config-file=/usr/local/percona/pmm2/config/pmm-agent.yaml > pmm-agent.log 2>&1 &
+            sleep 2
+          else
+            echo 'Setup installed PMM Client package...'
+            sudo pmm-admin config --force --server-insecure-tls --metrics-mode=auto --server-url=https://admin:admin@127.0.0.1:443
+            sudo usermod -a -G pmm-agent $USER
```
Comment (puneet0191) [inline .github/workflows/pmm2-testsuite.yml]: I am not able to understand what case would fall on this condition?

File: .github/workflows/pmm2-testsuite.yml [hunk 2]
```diff
@@ -127,7 +131,39 @@ jobs:
           sleep 30
 
       - name: Setup PMM2-Client
-        run: sudo -E bash -x ./pmm-tests/pmm2-client-setup.sh --pmm_server_ip 127.0.0.1 --client_version ${{ env.CLIENT_VERSION }} --admin_password admin --use_metrics_mode no
+        run: |
```
Comment (puneet0191) [inline .github/workflows/pmm2-testsuite.yml:135]: could we just have a separate GitHub reusable action that takes client_version as input parameter and sets up the client on GitHub worker?

**Review by puneet0191 — CHANGES_REQUESTED** [review]

File: .github/workflows/pmm2-testsuite.yml (hunk 1 above)
Comment (vyarosh) [reply .github/workflows/pmm2-testsuite.yml]: this condition is for package install - versions "empty", "dev-latest", "pmm2-rc" and  "pmm2-latest", this is the same way as in original script: https://github.com/percona/pmm-qa/blob/main/pmm-tests/pmm2-client-setup.sh#L101
I didn't find better structure to extract given closure without big duplications of code.
it's a setup fork: for tarball and not a tarball installation

File: .github/workflows/pmm2-testsuite.yml (hunk 2 above)
Comment (vyarosh) [reply .github/workflows/pmm2-testsuite.yml:135]: yes, we can, but how many? we have right now 4 install forks install "dev-latest", "pmm2-rc" and "pmm2-latest". And 2 setup forks: package setup and tarball setup. 
Also please note that this one is not final setup refactoring - it's the step intermediate state before moving it into "qa-integration"

Comment (vyarosh) [reply .github/workflows/pmm2-testsuite.yml:135]: updated with 2 setup steps

**Comment by vyarosh** [comment]
https://github.com/Percona-Lab/pmm-submodules/actions/runs/4447040158 - dev latest
https://github.com/Percona-Lab/pmm-submodules/actions/runs/4447041996/jobs/7808097089 - tarball

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-3139-bc1d712
client docker - perconalab/pmm-client-fb:PR-3139-bc1d712
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-3139-bc1d712.tar.gz
Create Staging Instance: https://pmm.cd.percona.com/job/aws-staging-start/parambuild/?DOCKER_VERSION=perconalab/pmm-server-fb:PR-3139-bc1d712&CLIENT_VERSION=https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-3139-bc1d712.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-3139-83dc6fe
client docker - perconalab/pmm-client-fb:PR-3139-83dc6fe
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-3139-83dc6fe.tar.gz
Create Staging Instance: https://pmm.cd.percona.com/job/aws-staging-start/parambuild/?DOCKER_VERSION=perconalab/pmm-server-fb:PR-3139-83dc6fe&CLIENT_VERSION=https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-3139-83dc6fe.tar.gz

File: .github/workflows/pmm2-testsuite.yml [hunk 3]
```diff
@@ -126,8 +130,32 @@ jobs:
...
+          sudo pmm-admin status
+
+      - name: Setup PMM2-Client tarball
+        if: startsWith(env.CLIENT_VERSION, 'http') || startsWith(env.CLIENT_VERSION, '2.')
+        run: |
+          curl -O https://raw.githubusercontent.com/Percona-QA/package-testing/master/scripts/pmm2_client_install_tarball.sh
+          sudo bash -x pmm2_client_install_tarball.sh -v ${{ env.CLIENT_VERSION }}
+          echo 'Setup local pmm-agent...'
+          sudo pmm-agent setup --force --config-file=/usr/local/percona/pmm2/config/pmm-agent.yaml --server-address=127.0.0.1:443 --server-insecure-tls --server-username=admin --server-password=admin
+          sudo pmm-agent --config-file=/usr/local/percona/pmm2/config/pmm-agent.yaml > pmm-agent.log 2>&1 &
```
Comment (ademidoff) [inline .github/workflows/pmm2-testsuite.yml:156]: Does this even need the root user?

**Comment by vyarosh** [comment]
not relevant

**Pull request closed without merging**